test/test_parser.cpp

```diff
@@ -265,4 +265,75 @@ TEST_CASE("Sim region cleanup causes memory errors (run with ASAN)")
     auto& templ = doc->get_templates().back();
     auto sims = templ.get_simregions();
     CHECK(sims.size() == 3);
+}
+
+TEST_CASE("Struct int,int initialization")
+{
+    auto doc =
+        document_fixture{}.add_default_process().add_global_decl("const struct { int x; int y; } s = {1, 1};").parse();
+    REQUIRE(doc);
+    auto& errors = doc->get_errors();
+    CHECK(errors.size() == 0);
+    CHECK(doc->get_warnings().size() == 0);
+}
+
+TEST_CASE("Struct int,double initialization")
+{
+    auto doc = document_fixture{}
+                   .add_default_process()
+                   .add_global_decl("const struct { int x; double y; } s = {1, 1.0};")
+                   .parse();
+    REQUIRE(doc);
+    auto& errors = doc->get_errors();
+    CHECK(errors.size() == 0);
+    CHECK(doc->get_warnings().size() == 0);
+}
+
+TEST_CASE("Struct double,double initialization")
+{
+    auto doc = document_fixture{}
+                   .add_default_process()
+                   .add_global_decl("const struct { double x; double y; } s = {1.0, 1};")
+                   .parse();
+    REQUIRE(doc);
+    auto& errors = doc->get_errors();
+    CHECK(errors.size() == 0);
+    CHECK(doc->get_warnings().size() == 0);
+}
+
+TEST_CASE("Nested struct int,double initialization")
+{
+    auto doc =
+        document_fixture{}
+            .add_default_process()
+            .add_global_decl("typedef struct { int x; double y; } S; struct { S s1; S s2; } s = {{5,5.5},{2,2.25}};")
+            .parse();
+    REQUIRE(doc);
+    auto& errors = doc->get_errors();
+    CHECK(errors.size() == 0);
+    CHECK(doc->get_warnings().size() == 0);
+}
+
+TEST_CASE("Meta struct")
+{
+    auto doc = document_fixture{}
+                   .add_default_process()
+                   .add_global_decl("meta struct { int x; double y; } s = {1, 1.0};")
+                   .parse();
+    REQUIRE(doc);
+    auto& errors = doc->get_errors();
+    CHECK(errors.size() == 0);
+    CHECK(doc->get_warnings().size() == 0);
+}
+
+TEST_CASE("Struct meta field")
+{
+    auto doc = document_fixture{}
+                   .add_default_process()
+                   .add_global_decl("meta struct { int x; meta double y; } s = {1, 1.0};")
```

Comment:
It would be nice to add more tests that are failing (test to fail):
- initializing `int` with floating point constant
- declaring `meta` field inside non-meta `struct` (I am not sure what semantics should be in such cases)

Then some extra tests to pass:
- initialize `struct` of `struct`s with mixed integers and decimals
- initialize `struct` of arrays with mixed integers and decimals
- initialize array of `struct`s with mixed integers and decimals